File: python/leetcode-75/arrays-and-strings/increasing-triplet-subsequence.py
from typing import List
import math

def increasingTriplet(nums: List[int]) -> bool:
        
        if len(nums) < 3: return False

        min1 = math.inf
        min2 = math.inf

        for num in nums:
            if num <= min1:
                min1 = num
            elif num <= min2:
                min2 = num
            else: return True

        return False

print(increasingTriplet([1,2,3,4,5]))

print(increasingTriplet([5,4,3,2,1]))

print(increasingTriplet([2,1,5,0,4,6]))

File: javascript/leetcode-75/arrays-and-strings/increasing-triplet-subsequence.js
var increasingTriplet = function(nums) {
  
    // return true if there are 3 numbers in the array from left to right 
    // that increase in value

    // we solve this by assigning any number lower than the max number
    // to min1 and then min2 and if these are met then a higher number
    // is found the loop terminates returning true and if the loop 
    // completes without satisfying these conditions then it returns false

    // handles the edge case of array with less than 3 values
    if (nums.length < 3) return false

    // this is the maximum allowable int in javascript 
    let min1 = Number.MAX_SAFE_INTEGER
    let min2 = Number.MAX_SAFE_INTEGER

    // can also use Infinity
    // let min1 = Infinity
    // let min2 = Infinity
    
    // iterates using num of nums, num in nums will error out
    for (let num of nums) {

        // assigns the lowest value in array
        if (num <= min1) {
            min1 = num;
        } 
        
        // assigns the 2nd lowest value in array
        else if (num <= min2) {
            min2 = num;
        }

        // returns true if num is greater than the lower 2 numbers min1 and min2
        else return true
    }

    // return false if the loop completes without finding a number higher than min2
    return false
}

console.log(increasingTriplet([1,2,3,4,5]))

console.log(increasingTriplet([5,4,3,2,1]))

console.log(increasingTriplet([2,1,5,0,46]))